client/ayon_core/hosts/houdini/api/lib.py

```diff
@@ -370,6 +371,16 @@ def imprint(node, data, update=False):
         if value is None:
             continue
 
+        if labeler is not None:
+            if not key.startswith(KEY_PREFIX):
+                key = KEY_PREFIX + key
+
+        # Use the key with the prefix by default.
+        # But, if that key doesn't exist, use the old key.
+        key_without_prefix = key.replace(KEY_PREFIX, "", 1)
+        if node.parm(key_without_prefix):
+            key = key_without_prefix
```

Comment:
So the way I proposed it in the comments was to make the `labeler` argument be a `function` so that you can imprint with any custom labeling making the imprint function much more re-usable for custom things.

Note that this function is inside `lib` - it should be as generic as possible if we can do so. As such this is also not the place to do any magic parm remapping.

Also, to swap the prefix you do not want to use `str.replace()` because it's not the same and may result in unexpected behavior, e.g. `my_ayon_parm` would suddenly become `my_parm` even though you wouldn't expect it.

From Py3.9 there is `str.removeprefix`. @iLLiCiTiT might know whether we still need to support older?
Otherwise do:
```python
if text.startswith(prefix):
    text = text[len(prefix):]
```

---

Side note, don't forget to update the docstring as well.

Reply:
I'll make it quick.
- I don't think `imprint` should use the prefix by default, it should be possible to use imprint for other data that are not AYON related.
- Ideally this PR should affect only instances from create plugins, if possible.
- Don't use `replace`, use `text = text[len(prefix):]`.
- Keep in mind that some places in code might access the parameters directly without using read functions and the function should know what the parameter name is.